packages/node_modules/node-red/red.js

```diff
@@ -143,7 +143,46 @@ if (process.env.NODE_RED_ENABLE_PROJECTS) {
 }
 
 if (settings.https) {
-    server = https.createServer(settings.https,function(req,res) {app(req,res);});
+    var startupHttps = settings.https;
+    
+    if (typeof startupHttps === "function") {
+        // Get the result of the function, because createServer doesn't accept functions as input
+        startupHttps = startupHttps(); 
+    }  
+
+    server = https.createServer(startupHttps,function(req,res) {app(req,res);});
+    
+    // Refresh https settings at intervals for NodeJs version 11 and above
+    if (settings.httpsRefreshInterval) {
+        if (typeof startupHttps === "function") {        
+            if (server.setSecureContext) {
+                console.log("Refreshing https settings every " + parseInt(settings.httpsRefreshInterval) + " mseconds.");
+                setInterval(function () {
+                    try {
+                        // Get the result of the function, because createServer doesn't accept functions as input
+                        var refreshedHttps = settings.https();
+                        
+                        if (!refreshedHttps.key || !refreshedHttps.cert) {
+                            console.log("Cannot refresh the https settings when the https property function doesn't return a 'key' and 'cert'.");
+                            return;
+                        }
+                            
+                        // Only update the credentials in the server when key or cert has changed
+                        if(!server.key || !server.cert || !server.key.equals(refreshedHttps.key) || !server.cert.equals(refreshedHttps.cert)) {
+                            server.setSecureContext(refreshedHttps);
+                            console.log("The https settings have been refreshed.");
+                        }
+                    } catch(err) {
+                        console.log("Failed to refresh the https settings: " + err);
+                    }
+                }, parseInt(settings.httpsRefreshInterval));
```

Comment:
OK - I don't mind if it's seconds as long as it is clear and obvious. 
Do we need any validation checks around it in case it's set badly ? eg to a string or NaN. ?

Reply:
I absolutely agree that input should be validated to avoid mistakes.  However it seems to me that other data from the setting.js file (e.g. ```settings.uiPort```...) also isn't validated.  So I'm not sure if I have to be more catholic than the pope ;-)